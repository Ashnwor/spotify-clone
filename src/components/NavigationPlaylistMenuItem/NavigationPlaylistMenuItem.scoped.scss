$color-white-active: #fff;
$color-white-inactive: #b3b3b3;
$color-navbar-black: #040404;

.playlist-menu-item {
  display: flex;
  width: 100%;
  height: 42px;
  align-items: center;
  background: $color-navbar-black;
  cursor: pointer;

  &:hover .icon-container {
    opacity: 1;
    transition: opacity 200ms linear;
  }

  &:hover .text {
    color: $color-white-active;
    transition: color 200ms linear;
  }
}

.icon-container {
  display: flex;
  width: 32px;
  height: 32px;
  align-items: center;
  justify-content: center;
  background: $color-white-active;
  opacity: 0.7;
  transition: opacity 200ms linear;
}

.icon {
  width: 13.33px;
  height: 13.33px;
}

.white-color {
  color: $color-white-active;
}

.gradient {
  background: linear-gradient(135deg, #450af5, #c4efd9);
}

.text {
  height: 16px;
  margin-left: 16px;
  color: $color-white-inactive;
  font-family: 'Segoe UI';
  font-size: 12px;
  font-style: normal;
  font-weight: 700;
  line-height: 16px;
  transition: color 200ms linear;
}
