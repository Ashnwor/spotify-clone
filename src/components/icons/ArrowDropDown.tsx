function SvgArrowDropDown(props: React.SVGProps<SVGSVGElement>) {
  return (
    <svg width="1em" height="1em" viewBox="0 0 10 5" fill="none" {...props}>
      <path d="M0 0l5 5 5-5H0z" fill="currentColor" />
    </svg>
  );
}

export default SvgArrowDropDown;
