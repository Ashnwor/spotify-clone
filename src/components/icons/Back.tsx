import * as React from 'react';

function SvgBack(props: React.SVGProps<SVGSVGElement>) {
  return (
    <svg width="1em" height="1em" viewBox="0 0 11 17" fill="none" {...props}>
      <path d="M9.575 16.353L0 8.177 9.575 0l.595.697-8.758 7.48 8.758 7.48" fill="currentColor" />
    </svg>
  );
}

export default SvgBack;
