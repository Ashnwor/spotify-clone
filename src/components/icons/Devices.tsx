import * as React from 'react';

function SvgDevices(props: React.SVGProps<SVGSVGElement>) {
  return (
    <svg width="1em" height="1em" viewBox="0 0 22 17" fill="none" {...props}>
      <path
        d="M2 2.027h18V0H2C.9 0 0 .912 0 2.027v12.166c0 1.115.9 2.027 2 2.027h4v-2.027H2V2.028zM12 8.11H8v1.804c-.61.558-1 1.349-1 2.251 0 .902.39 1.693 1 2.25v1.805h4v-1.804c.61-.558 1-1.359 1-2.251 0-.892-.39-1.693-1-2.25V8.11zm-2 5.576c-.83 0-1.5-.68-1.5-1.521 0-.841.67-1.52 1.5-1.52s1.5.679 1.5 1.52-.67 1.52-1.5 1.52zm11-9.631h-6c-.5 0-1 .507-1 1.014v10.137c0 .507.5 1.014 1 1.014h6c.5 0 1-.507 1-1.014V5.07c0-.507-.5-1.014-1-1.014zm-1 10.138h-4v-8.11h4v8.11z"
        fill="currentColor"
      />
    </svg>
  );
}

export default SvgDevices;
