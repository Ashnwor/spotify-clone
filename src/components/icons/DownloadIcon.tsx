import * as React from 'react';

function SvgDownloadIcon(props: React.SVGProps<SVGSVGElement>) {
  return (
    <svg width="1em" height="1em" viewBox="0 0 15 15" fill="none" {...props}>
      <path
        fillRule="evenodd"
        clipRule="evenodd"
        d="M8 9.657V4H7v5.65L5.076 7.414l-.758.65 3.183 3.702 3.195-3.7-.758-.653L8 9.657zM7.5 0a7.5 7.5 0 100 15 7.5 7.5 0 000-15zm0 14A6.508 6.508 0 011 7.5C1 3.916 3.916 1 7.5 1S14 3.916 14 7.5 11.084 14 7.5 14z"
        fill="#B3B3B3"
      />
    </svg>
  );
}

export default SvgDownloadIcon;
