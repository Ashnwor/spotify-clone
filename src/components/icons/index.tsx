export { default as ArrowDropDown } from './ArrowDropDown'
export { default as Back } from './Back'
export { default as CardPause } from './CardPause'
export { default as CardPlay } from './CardPlay'
export { default as Devices } from './Devices'
export { default as DownloadIcon } from './DownloadIcon'
export { default as Forward } from './Forward'
export { default as Home2 } from './Home2'
export { default as HomeActive } from './HomeActive'
export { default as Home } from './Home'
export { default as LibraryActive } from './LibraryActive'
export { default as Library } from './Library'
export { default as Logo } from './Logo'
export { default as NextIcon } from './NextIcon'
export { default as PauseIcon } from './PauseIcon'
export { default as Pen } from './Pen'
export { default as PlayIcon } from './PlayIcon'
export { default as PlaylistHeartOutline } from './PlaylistHeartOutline'
export { default as PlaylistHeart } from './PlaylistHeart'
export { default as PlaylistPlus } from './PlaylistPlus'
export { default as Playlist } from './Playlist'
export { default as PreviousIcon } from './PreviousIcon'
export { default as RepeatIcon } from './RepeatIcon'
export { default as SearchActive } from './SearchActive'
export { default as Search } from './Search'
export { default as ShuffleIcon } from './ShuffleIcon'
export { default as VolumeDown } from './VolumeDown'
export { default as VolumeMute } from './VolumeMute'
export { default as VolumeOff } from './VolumeOff'
export { default as VolumeUp } from './VolumeUp'