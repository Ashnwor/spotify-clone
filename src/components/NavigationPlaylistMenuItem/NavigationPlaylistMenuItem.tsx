import React from 'react';
import './NavigationPlaylistMenuItem.scss';

interface Props {
  text: string;
  Icon: React.FC<{ className: string }>;
  gradient?: boolean;
}

const NavigationPlaylistMenuItem: React.FC<Props> = ({ Icon, text, gradient }) => {
  return (
    <div className="btn-container-playlist-item">
      <div className="playlist-menu-inner-frame">
        <div
          className={`playlist-menu-icon-container ${gradient && 'playlist-menu-icon-gradient'}`}
        >
          <Icon className={`playlist-menu-icon ${gradient ? 'white-color' : ''}`} />
        </div>
        <span className="playlist-menu-text">{text}</span>
      </div>
    </div>
  );
};

export default NavigationPlaylistMenuItem;
