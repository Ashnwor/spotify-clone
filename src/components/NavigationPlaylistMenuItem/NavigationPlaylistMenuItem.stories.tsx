import React from 'react';
import NavigationPlaylistMenuItem from './NavigationPlaylistMenuItem';
import { PlaylistHeart, PlaylistPlus } from '../icons';

export default {
  title: 'NavigationPlaylistMenuItem',
};

export const navigationPlaylistMenuItem = () => {
  return (
    <div>
      <NavigationPlaylistMenuItem text="Create a playlist" Icon={PlaylistPlus} />
      <NavigationPlaylistMenuItem text="Liked Songs" Icon={PlaylistHeart} gradient />
    </div>
  );
};
