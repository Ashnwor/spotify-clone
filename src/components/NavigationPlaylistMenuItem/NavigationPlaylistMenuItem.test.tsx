import React from 'react';
import { shallow } from 'enzyme';
import NavigationPlaylistMenuItem from './NavigationPlaylistMenuItem';
import { PlaylistHeart } from '../icons';

it('runs correctly', () => {
  const wrapper = shallow(
    <NavigationPlaylistMenuItem text="test_text" Icon={PlaylistHeart} gradient />,
  );
  expect(wrapper).toMatchSnapshot();
});
