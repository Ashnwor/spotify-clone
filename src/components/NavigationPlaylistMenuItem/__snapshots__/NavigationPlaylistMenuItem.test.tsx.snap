// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`runs correctly 1`] = `
<div
  className="playlist-menu-item"
>
  <div
    className="icon-container gradient"
  >
    <SvgPlaylistHeart
      className="icon white-color"
    />
  </div>
  <span
    className="text"
  >
    test_text
  </span>
</div>
`;
